src/views/components/selects/SelectNature.tsx

```diff
@@ -1,30 +1,105 @@
 import { State, useGlobalState } from '@src/GlobalStateProvider';
-import { useTranslation } from 'react-i18next';
+import { TFunction, useTranslation } from 'react-i18next';
 import { SelectDataProps } from './SelectDataProps';
 import { getText } from '@utils/ReadingProjectText';
 import { SelectCustom } from '@components/SelectCustom';
 
-const getNatureOptions = (state: State): SelectOption[] =>
-  Object.entries(state.projectConfig.natures.db_symbol_to_id).map(([value, natureId]) => ({
-    value,
-    label: getText(
+/**
+ * Get the traduction by index
+ * @param index
+ * @param t
+ * @returns string
+ */
+const getTraductionByIndex = (index: number, t: TFunction<['database_natures']>) => {
+  switch (index) {
+    case 1:
+      return t('database_natures:attack');
+    case 2:
+      return t('database_natures:defense');
+    case 3:
+      return t('database_natures:speed');
+    case 4:
+      return t('database_natures:special_attack');
+    case 5:
+      return t('database_natures:special_defense');
+    default:
+      return '';
+  }
+};
+
+/**
+ * Get the nature extra stats compared to depending stats
+ * @param state
+ * @param t useTranslation
+ * @returns Record<string, string>
+ */
+const getNatureExtraStatsComparedToDependingStats = (state: State, t: TFunction<['database_natures']>): Record<string, string> => {
+  const natures = state.projectConfig.natures.db_symbol_to_id;
+  const stats = state.projectConfig.natures.data;
+
+  const natureExtraStatsComparedToDependingStats: Record<string, string> = {};
+
+  for (const [natureSymbol, natureId] of Object.entries(natures)) {
+    const natureStats = stats[natureId];
+    if (natureStats) {
+      // Get the index of the stats
+      const upStatIndex = natureStats.findIndex((ns, index) => index > 0 && ns > 100);
+      const downStatIndex = natureStats.findIndex((ns, index) => index > 0 && ns < 100);
+      // Get the traduction of the stats
+      const upStatTraduction = getTraductionByIndex(upStatIndex, t);
+      const downStatTraduction = getTraductionByIndex(downStatIndex, t);
+
+      // Constrution of the string
+      if (upStatTraduction) {
+        natureExtraStatsComparedToDependingStats[natureSymbol] = `+${upStatTraduction || ''}`;
+      }
+
+      if (downStatTraduction) {
+        natureExtraStatsComparedToDependingStats[natureSymbol] = `${upStatTraduction ? '+' + upStatTraduction : ''}
+        ${upStatTraduction ? ' / ' : ''}
+        ${downStatTraduction ? '-' + downStatTraduction : ''}`;
+      }
+    }
+  }
+  return natureExtraStatsComparedToDependingStats;
+};
+
+/**
+ * Get the nature options
+ * @param state
+ * @param t useTranslation
+ * @returns SelectOption[]
+ */
+const getNatureOptions = (state: State, t: TFunction<['database_natures']>): SelectOption[] =>
+  Object.entries(state.projectConfig.natures.db_symbol_to_id).map(([value, natureId]) => {
+    const test = getNatureExtraStatsComparedToDependingStats(state, t);
```

Comment:
You should not call this function on each iteration as it's an expensive one and it does not depend on any parameters of the iteration.

Add curly brackets to getNatureOptions and return the Object.entries :)

Reply:
Ah yes, did'nt see i was iterate multiple time on it, thanks for review !